homeassistant/components/rabbitair/__init__.py

```diff
@@ -0,0 +1,76 @@
+"""The Rabbit Air integration."""
+from __future__ import annotations
+
+import asyncio
+from datetime import timedelta
+import logging
+
+from rabbitair import Client, State, UdpClient
+
+from homeassistant.components import zeroconf
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_ACCESS_TOKEN,
+    CONF_HOST,
+    CONF_SCAN_INTERVAL,
+    Platform,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DEFAULT_SCAN_INTERVAL, DOMAIN, KEY_COORDINATOR, KEY_DEVICE
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS: list[Platform] = [Platform.FAN]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Rabbit Air from a config entry."""
+
+    hass.data.setdefault(DOMAIN, {})
+
+    host: str = entry.data[CONF_HOST]
+    token: str = entry.data[CONF_ACCESS_TOKEN]
+
+    _LOGGER.debug("Initializing with host %s, token %s", host, token[:5])
+    zeroconf_instance = await zeroconf.async_get_async_instance(hass)
+    device: Client = UdpClient(host, token, zeroconf=zeroconf_instance)
+
+    scan_interval: int = entry.options.get(CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL)
```

Comment:
Thanks for the tip! The changes were applied directly from options flow, but it is better to reload the config entry. In the future, more options will be added for which it will be difficult to apply changes on the fly, without reloading.